worker/caasoperator/caasoperator_test.go

```diff
@@ -133,11 +141,15 @@ func (s *WorkerSuite) TestStartStop(c *gc.C) {
 }
 
 func (s *WorkerSuite) TestWorkerDownloadsCharm(c *gc.C) {
+	ctx := &hookObserver{}
+	s.config.NewRunnerFactoryFunc = newRunnerFactoryFunc(ctx)
 	w, err := caasoperator.NewWorker(s.config)
 	c.Assert(err, jc.ErrorIsNil)
 	defer workertest.CleanKill(c, w)
 
 	s.settingsChanges <- struct{}{}
+	ctx.waitForHooks(c, []string{"config-changed"})
```

Comment:
why? what's this got to do with the test?

Reply:
we need to wait for the hook to fire or else the method calls get the "APIAddresses" called when setting up the hook context